spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/convert/SpannerConverters.java

```diff
@@ -143,17 +146,44 @@ public LocalDateTime convert(Timestamp timestamp) {
 	// @formatter:off
 	public static final Converter<java.util.Date, Timestamp>
 			DATE_TIMESTAMP_CONVERTER = new Converter<java.util.Date, Timestamp>() {
+		final java.util.Date commitTimestamp = CommitTimestamp.of(java.util.Date.class);
 		// @formatter:on
 		@Nullable
 		@Override
 		public Timestamp convert(java.util.Date date) {
-			long time = date.getTime();
-			long secs = Math.floorDiv(time, 1000L);
-			int nanos = Math.toIntExact((time - secs * 1000L) * 1000000L);
-			return Timestamp.ofTimeSecondsAndNanos(secs, nanos);
+			if (commitTimestamp == date) {
+				return Value.COMMIT_TIMESTAMP;
+			}
+			return toTimestamp(date);
 		}
 	};
 
+	/**
+	 * A converter from {@link java.util.Date} to the Spanner timestamp type.
+	 */
+	// @formatter:off
+	public static final Converter<java.sql.Date, Timestamp>
+			SQL_DATE_TIMESTAMP_CONVERTER = new Converter<java.sql.Date, Timestamp>() {
```

Comment:
Do we really need this new converter? Seems like it would be the subject of another PR.

Reply:
There is some problem with such "code" converters - no one can overwrite them because of initialization restrictions...
Basically I agree, I'll think about how to improve it better.